### ISSUE TITLE
Bug of matching absolute paths for some patterns.

### ISSUE THREAD

**Issue by Isaac0616** [cpburnz/python-path-specification#46]
Some patterns work with both absolute and relative paths.
For example,
```
>>> list(pathspec.patterns.GitWildMatchPattern("*.py").match(["/foo/a.py", "foo/a.py", "x/foo/a.py", "/x/foo/a.py"]))
['/foo/a.py', 'foo/a.py', 'x/foo/a.py', '/x/foo/a.py']
```
```
>>> list(pathspec.patterns.GitWildMatchPattern("**").match(["/foo/a.py", "foo/a.py", "x/foo/a.py", "/x/foo/a.py"]))
['/foo/a.py', 'foo/a.py', 'x/foo/a.py', '/x/foo/a.py']
```
However, the pattern `foo` or `/foo` won't match the path starts with `/foo`.
For example,
```
>>> list(pathspec.patterns.GitWildMatchPattern("foo").match(["/foo/a.py", "foo/a.py", "x/foo/a.py", "/x/foo/a.py"]))
['foo/a.py', 'x/foo/a.py', '/x/foo/a.py']
>>> list(pathspec.patterns.GitWildMatchPattern("/foo").match(["/foo/a.py", "foo/a.py", "x/foo/a.py", "/x/foo/a.py"]))
['foo/a.py']
```
Can we support matching the absolute path in this case?
I think making `output.append('(?:.+/)?')` at [here](https://github.com/cpburnz/python-path-specification/blob/7b125acf41702cb82679dcf56aaf6a14d34bd785/pathspec/patterns/gitwildmatch.py#L132) ``output.append('(?:.*/)?')`` could solve the issue, but I am not sure whether it has other unwanted side effect.
Another solution is that we normalize `/...` to `...` similar to normalizing the `./` at [here](https://github.com/cpburnz/python-path-specification/blob/7b125acf41702cb82679dcf56aaf6a14d34bd785/pathspec/util.py#L327).

**Comment by cpburnz** [comment]
Thank you for the bug report. This is a normalization issue which is now fixed and will be in the next release.

**Comment by cpburnz** [comment]
This improvement has been released in v0.9.0.
